lib/literature/migrations/postgres.ex

```diff
@@ -0,0 +1,98 @@
+defmodule Literature.Migrations.Postgres do
```

Comment:
@jeffmartide I believe we are overcomplicating this 

https://hexdocs.pm/ecto_sql/Ecto.Migrator.html#module-example-running-migrations-in-a-release

We can put versions on the migrations to trigger specific versions an run it on release. 

```
mix ecto gen.migration 123213213_add_literature_v1.exs
mix ecto gen.migration 000000_add_more_literuate_v2.exs


 Ecto.Migrator.with_repo(repo, &Ecto.Migrator.run(&1, :up, all: true)
 
 This will trigger up on any pending migration in the release. I'm not sure how it will work in the 
 timestamp we currently have in our system but this should be the way we do migrations going forward
```

Reply:
wouldn't this trigger always the latest when elixir runs `.up()` or `.down()`? it may fail migration if we dont specify the exact version.

Reply:
@jeffmartide we have a database table called "schema_migrations" which stores  the version and inserted_at 

```
 20170417030534 | 2023-02-16 15:55:11
 20170417030535 | 2023-02-16 15:55:11
 20170417034410 | 2023-02-16 15:55:11
 20170418142256 | 2023-02-16 15:55:11
 20170418143136 | 2023-02-16 15:55:11
```

What this function does in the release module is it hooks into Martide.Repo which can be set in the config of literature and triggers ecto_repo to run all pending migrations. A pending migration is considered not run yet if the schema_migrations table is missing the version. What this means is it will work similar to how migrations work in our app. It ignores the timestamps already inserted in the version and creates the migrations which are pending.

We will run into issues though as we need to fix  how we ran the previous v1 migration but if we fix that then this becomes alot easier to update add and create new schemas

```
 Ecto.Migrator.with_repo(Martide.Repo, &Ecto.Migrator.run(&1, :up, all: true)
```